nodebuilder/share/constructors.go

```diff
@@ -21,17 +21,17 @@ import (
 	"github.com/celestiaorg/celestia-node/share/getters"
 )
 
-func discovery(cfg Config) func(routing.ContentRouting, host.Host) *disc.Discovery {
+func newDiscovery(cfg Config) func(routing.ContentRouting, host.Host) *disc.Discovery {
 	return func(
 		r routing.ContentRouting,
 		h host.Host,
 	) *disc.Discovery {
 		return disc.NewDiscovery(
 			h,
 			routingdisc.NewRoutingDiscovery(r),
-			cfg.PeersLimit,
-			cfg.DiscoveryInterval,
-			cfg.AdvertiseInterval,
+			disc.WithPeersLimit(cfg.Discovery.PeersLimit),
+			disc.WithDiscoveryInterval(cfg.Discovery.DiscoveryInterval),
+			disc.WithAdvertiseInterval(cfg.Discovery.AdvertiseInterval),
```

Comment:
Don't see logic that uses CacheAvailability options anywehre

Reply:
Agreed, in this PR we open it for configuration (_specifically and only, `WriteBatchSize`_) just in case somewhere down the line we might need to play with this in tests or something.

Reply:
Hmmm. not so sure about this. YAGNI